firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/Serial/Nonin/SensorAlarmService.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * SensorAlarmService.cpp
+ *
+ *  Created on: June 6, 2020
+ *      Author: Ethan Li
+ */
+
+#include "Pufferfish/Driver/Serial/Nonin/SensorAlarmService.h"
+
+namespace Pufferfish::Driver::Serial::Nonin {
+
+using Application::LogEventCode;
+using Application::LogEventType;
+const LogEventType system = LogEventType::LogEventType_system;
+
+void SensorAlarmsService::transform(
+    InitializableState status,
+    const SensorConnections &sensor_connections,
+    Application::AlarmsManager &alarms_manager) {
+  if (status == InitializableState::failed) {
+    alarms_manager.activate_alarm(LogEventCode::LogEventCode_nonin_disconnected, system);
+    return;
+  }
+
+  if (status != InitializableState::failed) {
```

Comment:
Yes so when the `status != ...:failed`, we would still need to deactivate the alarm right? since we are returning early if it's failed and not checking any other `status codes`